src/main/java/com/aws/greengrass/clientdevices/auth/iot/GreengrassV2DataClientFactory.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot;
+
+import com.aws.greengrass.componentmanager.ClientConfigurationUtils;
+import com.aws.greengrass.deployment.DeviceConfiguration;
+import com.aws.greengrass.util.Coerce;
+import com.aws.greengrass.util.GreengrassServiceClientFactory;
+import com.aws.greengrass.util.Utils;
+import software.amazon.awssdk.auth.credentials.AnonymousCredentialsProvider;
+import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+import software.amazon.awssdk.core.retry.RetryPolicy;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.greengrassv2data.GreengrassV2DataClient;
+import software.amazon.awssdk.services.greengrassv2data.GreengrassV2DataClientBuilder;
+
+import java.net.URI;
+import javax.inject.Inject;
+
+/**
+ * Provides GG service clients built specifically for CDA use case.
+ * Ideally this factory should maintain the lifecycle of its clients,
+ * but for now its upto the consumer of the clients to maintain their lifecycle.
+ */
+public class GreengrassV2DataClientFactory {
+    private final GreengrassServiceClientFactory greengrassServiceClientFactory;
+
+    /**
+     * Construct Greengrass Client Factory.
+     *
+     * @param greengrassServiceClientFactory GG Service Client Factory to retrieve the latest device-config
+     */
+    @Inject
+    public GreengrassV2DataClientFactory(GreengrassServiceClientFactory greengrassServiceClientFactory) {
```

Comment:
why are we injecting the factory? That seems pretty weird. You're only using it to get the device configuration, so why aren't we just injecting the `DeviceConfiguration`?

Reply:
`GreengrassServiceClientFactory` subscribes to `DeviceConfiguration` updates and also validates those updates. We're injecting this factory here to avoid having to subscribe and validate device config updates.

Reply:
Why would you need to do that?

The original client factory will still do its work whether or not you inject it here.

Reply:
Let's say that the factory finds the config to be invalid. This code in this class doesn't do anything differently, it doesn't know or care about the validation because that code isn't here.

Reply:
Makes sense, fixed it now.